### ISSUE TITLE
HBM causing Cascading worldgen lag

### ISSUE THREAD

**Issue by VanLabyrinth** [Alcatergit/Hbm-s-Nuclear-Tech-GIT#424]
When generating new terrain i get this error:
_[05:12:31] [Server thread/WARN] [FML]: HBM's Nuclear Tech - Extended Edition loaded a new chunk [CHUNK CORDS] in dimension 0 (overworld) while populating chunk [-243, -736], causing cascading worldgen lag.
[05:12:31] [Server thread/WARN] [FML]: Please report this to the mod's issue tracker. This log can be disabled in the Forge config._

I understand that this is a problem with the mods code, but if you do see this could you please supply some more insight into the problem? 
thank you

**Comment by Xeredek** [comment]
Not a dev but my understanding on cascading worldgen is it can trigger when you have high spawn rates for any structure that overlaps multiple chunks.
so like- spawning a structure creates more chunks, which then triggers more structures with more chunks ect and you get a cascade where the game is just sitting there spawning further and further chunks ad infinitum.
This is unfortunately kind of a fundamental flaw with MC, again to my limited understanding. 

if your tune down generating structures generally (which includes hmb cause the default values are bonkers) it should clear up at some setting

**Comment by VanLabyrinth** [comment]
Yeah i did have a few other structure mods in, and did find when i turned
down their spawn rates the crashes were alot less frequent

**Comment by Shirhix** [comment]
I don't understand why this can't be fixed though. Like, just don't generate part of the structure that is not to be loaded.
Just a bit of math is it not?
